rpcd/playbooks/roles/rpc_maas/library/raxmon.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ConfigParser
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+DOCUMENTATION = """
+---
+module: raxmon
+short_description:
+  - A module for preparing a node for Rackspace Cloud Monitoring
+description:
+  - This module is written for rpc-openstack's rpc_maas role specifically to
+    prepare a node for Rackspace Cloud Monitoring by a) assigning an agent
+    to an entity and b) creating an agent token.
+options:
+  cmd:
+    description: The command to run
+    choices = [ 'assign_agent_to_entity', 'create_agent_token' ]
+    required: true
+  entity:
+    description: The label of the entity to operate against
+    required: true
+  venv_bin:
+    description: The path to the venv where rackspace-monitoring is installed
+    required: false
+  raxmon_cfg:
+    description: The path to the raxmon configuration file
+    required: false
+    default: /root/.raxrc
+"""
+
+EXAMPLES = """
+raxmon:
+  cmd: assign_agent_to_entity
+  entity: controller1
+  venv_bin: /openstack/venvs/maas-r14.1.0rc1/bin/
+
+raxmon:
+  cmd: create_agent_token
+  entity: controller1
+  venv_bin: /openstack/venvs/maas-r14.1.0rc1/bin/
+"""
+
+RETURN = """
+id:
+  description:
+    - The ID of the entity's agent token, returned when cmd=create_agent_token
+  returned: success
+  type: id
+"""
+
+
+def _get_agent_tokens(conn, entity):
+    agent_tokens = []
+    for a in conn.list_agent_tokens():
+        if a.label == entity:
+            agent_tokens.append(a)
+    return agent_tokens
+
+
+def _get_conn(get_driver, Provider, raxmon_cfg):
```

Comment:
It may be personal preference, but I think vars you define to be passed in should be lower case.  Also this variable shadows the outer import, which is bad.